sync.js

```diff
@@ -150,8 +156,9 @@ class SyncState {
 
   onprogress (peer, progress) {
     if (this._isclosed(peer)) return
-    progress.timestamp = Date.now()
-    peer.state = PeerState(ReplicationState.PROGRESS, progress)
+    var newProgress = Object.assign({}, progress)
```

Comment:
If I recall correctly, we added this because we thought that the timestamp wasn't changing, and then confirmed it was. I think we can remove this copying logic safely.